pandas/core/dtypes/dtypes.py

```diff
@@ -896,7 +896,8 @@ def __new__(cls, freq=None):
 
         elif freq is None:
             # empty constructor for pickle compat
-            u = object.__new__(cls)
+            # -10_000 corresponds to PeriodDtypeCode.UNDEFINED
```

Comment:
is this actually hit in tests? what is the point of UNDEFINED? I mean, when does this become defined

Reply:
Yes.  We have a test that constructs `PeriodDtype()` which goes through this path.  the existing comment a couple lines up suggests this is for pickle compat

Reply:
right, how does this get resolved though? do you see periods with undefined after this?

Reply:
i think the idea is to use it as a placeholder until you can get a "real" PeriodDtype.  e.g. if you know that you are parsing a csv column to Period but dont yet know the freq.  just speculating here